### ISSUE TITLE
[BUG] [Custom Controls] Layout scaling inaccuracy

### ISSUE THREAD

**Issue by Felix14-v2** [PojavLauncherTeam/PojavLauncher#4706]
### Description

When creating my custom Bedrock–like controls, I ran into this strange problem - PojavLauncher incorrectly calculates button positions (or their size). This becomes well noticeable when you mess with a small buttons, but also may affect the large ones. I don't know the exact reason, but as a result we have this annoying problem – there are always small inaccuracies in the vertical positioning of the buttons (you can see horizontal gaps in all screenshots), and significant inaccuracies in horizontal positioning at certain scale values. All "pixels" are aligned with mathematical precision, this isn't an issue of position expressions I'm using.

### Demo

<details>
 <summary>Screenshots (9)</summary>

80%
![image](https://github.com/PojavLauncherTeam/PojavLauncher/assets/75726196/bd349767-c7b5-42c4-957d-e5da97530328)

100%
![image](https://github.com/PojavLauncherTeam/PojavLauncher/assets/75726196/8a885770-dfd7-4f5e-bb3f-33f456c2f35d)

125%
![image](https://github.com/PojavLauncherTeam/PojavLauncher/assets/75726196/089b285b-125b-48d0-8369-05df96df5774)

135%
![image](https://github.com/PojavLauncherTeam/PojavLauncher/assets/75726196/4de33879-a116-4d16-99ed-48888c46d204)

150%
![image](https://github.com/PojavLauncherTeam/PojavLauncher/assets/75726196/c5330f5b-7348-4bbd-a99e-fbd449978ef0)

165%
![image](https://github.com/PojavLauncherTeam/PojavLauncher/assets/75726196/eeddbcc2-f8c2-4995-b62a-1b11afe55159)

200%
![image](https://github.com/PojavLauncherTeam/PojavLauncher/assets/75726196/e146a6c8-17ff-41ac-975c-ff015c010d7b)

225%
![image](https://github.com/PojavLauncherTeam/PojavLauncher/assets/75726196/c8e7da2f-cf10-4d2a-b256-9ddce13115e5)

250%
![image](https://github.com/PojavLauncherTeam/PojavLauncher/assets/75726196/fdc4529b-5ef0-4e01-a28a-771b01e06bd9)

</details>

<details>
 <summary>With even larger scale (3)</summary>

![Screenshot_20231007-023344](https://github.com/PojavLauncherTeam/PojavLauncher/assets/75726196/114ed59a-0d10-4d4e-97e4-b3ca657ff707)
![Screenshot_20231007-023313](https://github.com/PojavLauncherTeam/PojavLauncher/assets/75726196/3d5e82e0-6f15-46a2-b721-b948781a35e2)
![Screenshot_20231007-023243](https://github.com/PojavLauncherTeam/PojavLauncher/assets/75726196/5f71a87c-ad99-468b-b468-eb0ed643f417)

</details>

### To Reproduce

```markdown
1. Import my test controls
2. Open it in the Pojav editor
3. Remember the state
4. Change the buttons scaling
5. Compare the result
6. See the difference
```

### Expected Behavior

Well, ideally I'd like to see a pixel-perfect alignment. Idk is it possible, but this problem stops me from creating cool stuff 😭

### Platform

```markdown
- Device model: all platforms 
- Android version: tested on 11 and 13 (WSA)
- PojavLauncher version: latest from actions/GP
```

### Additional

[Controls for testing](https://cdn.discordapp.com/attachments/902664003736707093/1159989008034697266/For_testing.json?ex=680e0eb3&is=680cbd33&hm=6a6d44b76b6f5ed65f741185706f2ab4d6687d36428f6dad1a7b53646741c9fc&)

**Comment by Felix14-v2** [comment]
I found that horizontal gaps only occur when you set the Y position using ${screen_height} (or ${bottom}) variable.

Here the button is bound to the top of the screen,
![image](https://github.com/PojavLauncherTeam/PojavLauncher/assets/75726196/fe906e51-9a71-40e4-9de0-265584f0f31a)

... and here – to the bottom
![image](https://github.com/PojavLauncherTeam/PojavLauncher/assets/75726196/64d0eab8-a83f-43fa-8455-f268f3323388)

UPD: this exact case needs more investigating since I can't reproduce it with other buttons

**Comment by artdeell** [comment]
Btw, i'd imagine that doing buttons this way would have an insane lag
cost...

вс, 8 окт. 2023 г., 18:02 Felix14_v2 ***@***.***>:

**Comment by Felix14-v2** [comment]
Yeah, first of all I tested the limits, my MediaTek G85 can easily withstand about 500 buttons without input lag. But anyways the button from the screenshots above is just proof-of-concept, just to say "I recreated Bedrock controls, don't use it" :D

But the idea of constructing a button design of multiple elements is still used in my main controls. For example, this button
![image](https://github.com/PojavLauncherTeam/PojavLauncher/assets/75726196/4b324ace-2a9f-488a-9cd4-2c172d6664d9) 
... consists of only 23 elements, and the button from the previous message can be made from only 17 (currently 151). So I still far from the limit

**Comment by Felix14-v2** [comment]
> [Controls for testing](https://cdn.discordapp.com/attachments/902664003736707093/1159989008034697266/For_testing.json?ex=680e0eb3&is=680cbd33&hm=6a6d44b76b6f5ed65f741185706f2ab4d6687d36428f6dad1a7b53646741c9fc&)
I can no longer reproduce this bug. Can someone test this to confirm that the bug is fixed?
